src/lazy/index.tsx

```diff
@@ -21,41 +21,49 @@ export type Options = {
   moduleId?: string;
 };
 
-type LazyComponent = React.ComponentType<any> & {
+export type LazyComponent<P> = ComponentType<P> & {
   preload: () => void;
   getAssetUrls: (manifest: Manifest) => string[] | undefined;
 };
 
-const lazyProxy = (
-  loader: Loader,
+function lazyProxy<P>(
+  loader: Loader<P>,
   { defer = PHASE.PAINT, moduleId = '', ssr = true }: Options = {}
-): LazyComponent => {
+): LazyComponent<P> {
   const isServer = isNodeEnvironment();
   const dataLazyId = hash(moduleId);
 
-  const LazyComponent: any = isServer
+  const LazyComponent: ComponentType<P> = isServer
     ? createComponentServer({
         dataLazyId,
-        loader: loader as ServerLoader,
+        loader: loader as ServerLoader<P>,
         moduleId,
         ssr,
       })
     : createComponentClient({
         dataLazyId,
         defer,
-        deferred: createDeferred(loader as ClientLoader),
+        deferred: createDeferred(loader as ClientLoader<P>),
         moduleId,
         ssr,
       });
 
   LazyComponent.displayName = `Lazy(${displayNameFromId(moduleId)})`;
 
+  const getAssetUrls = (manifest: Manifest) => {
+    if (!manifest[moduleId]) {
+      return undefined;
```

Comment:
/nit

if you just return it's same as undefined:

Eg:
```js

if (!manifest[moduleId]) {
      return;
}
```

Reply:
Done